container.go

```diff
@@ -63,24 +63,23 @@ func (c *Container) Provide(t interface{}) error {
 	defer c.Unlock()
 
 	ctype := reflect.TypeOf(t)
-
 	switch ctype.Kind() {
 	case reflect.Func:
-		switch ctype.NumOut() {
-		case 2:
-			if ctype.Out(1) != _typeOfError {
-				return errReturnErrKind
-			}
-			fallthrough
-		case 1:
+		count := ctype.NumOut()
+		if count == 0 {
+			return errReturnCount
+		} else if count == 1 {
 			objType := ctype.Out(0)
 			if objType.Kind() != reflect.Ptr && objType.Kind() != reflect.Interface {
 				return errReturnKind
 			}
-		default:
-			return errReturnCount
+		} else {
+			// last variable must be error
+			if ctype.Out(count-1) != _typeOfError {
```

Comment:
I don't think we should ever force the error, right? Some objects never fail to construct and don't return the errors. In those cases we don't want to force people to wrap existing constructors just to return a nil here for error. 

I'd say check if the last one is an error, and if not consider them all objects for the graph.